src/backend/users/management/commands/migrate_housing_seeker_data.py

```diff
@@ -0,0 +1,488 @@
+
+        if has_voucher and has_rooms and has_priorities and has_modes:
+            voucher_number = profile[ProfileKeys.VOUCHER]
+            voucher_bedrooms = profile[ProfileKeys.ROOM]
+            travel_mode = self.decode_travel_mode(
+                profile[ProfileKeys.RAIL], profile[ProfileKeys.CAR]
+            )
+            commute_priority = self.decode_priority(profile[ProfileKeys.COMMUTE])
+            public_safety_priority = self.decode_priority(profile[ProfileKeys.SAFETY])
+            school_quality_priority = self.decode_priority(profile[ProfileKeys.SCHOOL])
+
+            if key in self.same_email_newer_voucher_keys:
+                self.stdout.write("User profile to be inserted is newer, updating...\n")
+                existing_profile = UserProfile.objects.get(user=user)
+                existing_profile.full_name = f"{user.first_name} {user.last_name}"
+                existing_profile.has_voucher = has_voucher
+                existing_profile.voucher_bedrooms = voucher_bedrooms
+                existing_profile.travel_mode = travel_mode
+                existing_profile.commute_priority = commute_priority
+                existing_profile.public_safety_priority = public_safety_priority
+                existing_profile.school_quality_priority = school_quality_priority
+                existing_profile.save()
+                user_profile = UserProfile.objects.get(user=user)
+            else:
+                user_profile = UserProfile(
+                    user=user,
+                    full_name=f"{user.first_name} {user.last_name}",
+                    has_voucher=has_voucher,
+                    voucher_number=voucher_number,
+                    voucher_bedrooms=voucher_bedrooms,
+                    travel_mode=travel_mode,
+                    commute_priority=commute_priority,
+                    public_safety_priority=public_safety_priority,
+                    school_quality_priority=school_quality_priority,
+                )
+                user_profile.save()
+                self.added_profiles += 1
+        else:
+            self.skipped_user_profile_key_s3.append({"key": key, "reason": reason})
+            self.stdout.write(f"{reason}. Skipping user profile with key: {key}.\n")
+
+        return user_profile
+
+    def has_location_fields(self, destination):
+        reason = ""
+        if DestinationKeys.LOC not in destination:
+            reason += "NO LOCATION; "
+            return False, reason
+        loc = destination[DestinationKeys.LOC]
+        if LocationKeys.POSITION not in loc:
+            reason += "NO POSITION; "
+            return False, reason
+        pos = loc[LocationKeys.POSITION]
+        if PositionKeys.LAT not in pos or PositionKeys.LON not in pos:
+            reason += "NO FULL COORDINATE PAIR; "
+            return False, reason
+
+        return True, reason
+
+    def decode_purpose(self, purpose):
+        if purpose.lower() == "friends":
```

Comment:
Not critical, but another way to set this up that I slightly prefer is something like this; it's slightly more compact but I don't think it sacrifices too much readability:

```python
PURPOSE_MAPPING = {
    "friends": Destination.TripPurpose.FRIENDS,
    "work": Destination.TripPurpose.WORK
    ...
}

def decode_purpose(self, purpose):
    return PURPOSE_MAPPING.get(purpose.lower(), Destination.TripPurpose.OTHER)
```

Reply:
Makes sense. Addressed here: https://github.com/azavea/echo-locator/pull/478/commits/a36a8f148ac0ae9312f701d016466514c1a7c864